### PR TITLE
Epmlstrtai 718 Move logic from service to entity (Token) 

### PR DESCRIPTION
**Pull request by glinko1567**
Created TokenTrait interface and moved some logic from TokenService and Token entity to it. Also added null check method for Token in UtilsService.

### CODE REVIEW

File: src/main/java/com/epam/test_generator/services/utils/UtilsService.java [hunk 1]
```diff
@@ -29,6 +25,17 @@
         }
     }
 
+    /**
+     * Checks if Token is null. Returns nothing if not.
+     *
+     * @throws TokenMissingException if Token is null.
+     */
+    public static void checkNotNullToken(Token token) {
```
Comment (pomkine) [inline src/main/java/com/epam/test_generator/services/utils/UtilsService.java]: Let's avoid using static methods like that one

Comment (glinko1567) [reply src/main/java/com/epam/test_generator/services/utils/UtilsService.java]: Is it ok then to check Token for null everywhere we use it? 
Or there is a better solution?

Comment (glinko1567) [reply src/main/java/com/epam/test_generator/services/utils/UtilsService.java]: Ok. Done.

File: src/main/java/com/epam/test_generator/entities/Token.java [hunk 2]
```diff
@@ -1,21 +1,22 @@
...
 import javax.persistence.OneToOne;
 import javax.validation.constraints.NotNull;
 
-
 /**
  * This class represents token essence. Token is a special key that is used for user identification.
  */
 @Entity
-public class Token {
+public class Token implements TokenTrait {
```
Comment (hdghg) [inline src/main/java/com/epam/test_generator/entities/Token.java]: Can you please fix javadoc by renaming "essence" to "entity", and adding cases what token is actually used for.

File: src/main/java/com/epam/test_generator/entities/Token.java [hunk 3]
```diff
@@ -24,44 +25,50 @@
...
-    public void setUser(User user) {
-        this.user = user;
-    }
-
-    public Token() {
+    @NotNull
+    private Date expiryDate;
 
+    public Token(Integer minutes) {
+        this.token = UUID.randomUUID().toString();
```
Comment (hdghg) [inline src/main/java/com/epam/test_generator/entities/Token.java]: IMO static builder would be more clean way to express what this piece of code performs.

    public static Token withExpiryDuration(int minutes) {
        Token result = new Token();
        result.token = UUID.randomUUID().toString();
        result.expiryDate = Date.from(Instant.now().plus(minutes, ChronoUnit.MINUTES));
        return result;
    }

File: src/main/java/com/epam/test_generator/entities/api/TokenTrait.java [hunk 4]
```diff
@@ -0,0 +1,15 @@
+package com.epam.test_generator.entities.api;
+
+import com.epam.test_generator.entities.Token;
+
+import java.util.Date;
+
+public interface TokenTrait {
+
+    Token getInstance();
```
Comment (hdghg) [inline src/main/java/com/epam/test_generator/entities/api/TokenTrait.java]: getInstance() should not be here. Instead, pull method  getExpiryDate.

File: src/main/java/com/epam/test_generator/entities/Token.java (hunk 2 above)
Comment (glinko1567) [reply src/main/java/com/epam/test_generator/entities/Token.java]: Fixed

File: src/main/java/com/epam/test_generator/entities/Token.java (hunk 3 above)
Comment (glinko1567) [reply src/main/java/com/epam/test_generator/entities/Token.java]: Done

File: src/main/java/com/epam/test_generator/entities/api/TokenTrait.java (hunk 4 above)
Comment (glinko1567) [reply src/main/java/com/epam/test_generator/entities/api/TokenTrait.java]: Done

**Pull request merged**